qiskit/transpiler/synthesis/graysynth.py

```diff
@@ -176,7 +176,7 @@ def graysynth(cnots, angles, section_size=2):
         else:
             sta.append([cnots1, list(set(ilist).difference([j])), qubit])
         sta.append([cnots0, list(set(ilist).difference([j])), qubit])
-    qcir += cnot_synth(state, section_size)
+    qcir += cnot_synth(state, section_size).inverse()
```

Comment:
This change seems to break `test.python.transpiler.test_synthesis.TestGraySynth.test_paper_example`.

Reply:
The circuit diagram in the paper was actually wrong in the first place. Seems to be an example of "two wrongs making a right"

Reply:
oh! a mistake in the paper... good catch.